### ISSUE TITLE
How can force replication regardless if there are snapshots only present on the receiver? 

### ISSUE THREAD

**Issue by ivhus** [zrepl/zrepl#543]
How can I always force replication of all the snapshots not present on the receiver? Regardless if there are some present only on the receiver.

Sorry if the question is stupid, I just didn't find the right configuration option for that in documentation.

**Comment by problame** [comment]
I don't understand your question. Unless there is a replication conflict, zrepl will replicate all snapshots created after the initial replication.

**Comment by InsanePrawn** [comment]
The way I understand it, this question seems to be asking for conflict resolution (#408), specifically for `receiver_only_snapshots: destroy`?

It's on our roadmap, but currently the only solution is to manually destroy the conflicting snapshots on the receiver.

**Comment by ivhus** [comment]
thanks for the reply

Let me rephrase since it wasn't clear: what I meant is — how can I force ignoring the fact that there may be some snapshots existing only the receiver (which does not create any problem) and just copy from sender to receiver all the snapshots which do not exists on the receiver.

> It's on our roadmap, but currently the only solution is to manually destroy the conflicting snapshots on the receiver.

I don't want to delete them, why not just ignore them and replicate all the rest.

There ara extra snapshots on the receiver — not an obstacle at all, right? It doesn't prevent copy all the rest from the sender.

I try to understand how should I configure zrepl to not look at the snapshots present only on receiver, leave them alone and just copy the rest.

**Comment by InsanePrawn** [comment]
> thanks for the reply
> 
> Let me rephrase since I wasn't clear: what I meant is — how can I force ignoring the fact that there may be some snapshots existing only on the receiver (their existance does not create any problem) and just copy from sender to receiver all the snapshots which do not exist on the receiver.
> 
> > It's on our roadmap, but currently the only solution is to manually destroy
> 
> I don't want to delete them, I expect just ignoring them and replicate all the rest.
> 
> There might be some extra snapshots created on the receiver — not an obstacle at all, right? It doesn't prevent copying of all the rest from the sender.

Sadly no, that is incorrect. ZFS allows no such thing.
Snapshots cannot be merged.
ZFS is not a clustered filesystem - you cannot modify (or even just snapshot) both sides of the replication. Sorry.
